### ISSUE TITLE
Error running jekyll

### ISSUE THREAD

**Issue by quanganhdo** [tbodt/ish#240]
I first installed jekyll using `apk add ruby`, followed by `gem install bundler jekyll`. The process failed with various errors.

![z1228456389872_05a25c8f4b6b33f7ba3706a99900e598](https://user-images.githubusercontent.com/18567/50580175-186d5f80-0e7e-11e9-9c30-139835e5225e.jpg)

I'm using build 40.

**Comment by tbodt** [comment]
The ruby package doesn't include all of ruby. Does it work better if you install ruby-full?

**Comment by quanganhdo** [comment]
It still failed with the same error (can't find header files for ruby. extconf failed) while building gem native extension for ruby_http_parser.

(I reinstalled ish and tried again with ruby-full)

**Comment by tbodt** [comment]
Just checked and for some reason even ruby-full isn't all of ruby. Try installing ruby-dev in addition?

**Comment by quanganhdo** [comment]
It worked with ruby-dev!

**Comment by deadbone** [comment]
> It worked with ruby-dev!

 can you use bundle exec jekyll serve ? Bundler et jekyll installs are fine for me but still getting an error with this command.

**Comment by tbodt** [comment]
ruby-full should include ruby-bigdecimal, do you have both ruby-dev and ruby-full installed?

**Comment by deadbone** [comment]
Ok I started a clean install with both ruby-dev and ruby-full

Now I’ve this error : 

```
DeadPadpro:~/blog# bundle exec jekyll serve                            
Configuration file: /root/blog/_config.yml                             
            Source: /root/blog                                         
       Destination: /root/blog/_site                                   
 Incremental build: disabled. Enable with --incremental                
      Generating...                                                    
       Jekyll Feed: Generating feed for posts                          
                    done in 16.944 seconds.                            
SEGV received in SEGV handler                                          
Aborted                                                                
DeadPadpro:~/blog# dmesg                                               
FIXME no support for shared futexes                                    
FIXME no support for shared futexes                                    
FIXME no support for shared futexes                                    
FIXME no support for shared futexes                                    
104 missing syscall 332                                                
104 page fault on 0xc at 0xf7e40db0                                    
104 page fault on 0xc at 0xf7e31deb                                    
DeadPadpro:~/blog#    
```

**Comment by tbodt** [comment]
That would be because support for file watching is missing.

**Comment by MattKevan** [comment]
I’m having a problem installing Jekyll - I’ve installed ruby, ruby-dev and ruby-full. Bundled installs fine, but installing Jekyll fails with

```make: Warning: File 'Makefile' has modification time 0.67 s in the future                                                                  
make: warning:  Clock skew detected.  Your build may be incomplete.```

Being able to run Jekyll locally would be fantastic.

**Comment by MattKevan** [comment]
You’re a genius!

As file watching support is missing, I managed to get Jekyll working by running  ``` bundle exec jekyll serve —no-watch ``` I’ll need to run the command again to see changes, but I can live with that.

For reference, I needed to install alpine-sdk, ruby-dev, ruby-full and libffi-dev to get the install to complete successfully. 

![751362e5-aea4-4846-8b67-25f99ac2b4e7](https://user-images.githubusercontent.com/1121942/50782162-33393d00-129f-11e9-93a2-b1d2f3285544.png)
